### PR TITLE
DE-1801 update actions core

### PR DESCRIPTION
**Pull request by djthornton1212**
# Pull Request Template

CI:

Jira: [DE-1801](https://caregilitycore.atlassian.net/browse/DE-1801)

## Description

This pull request is intended to update various components of this action to bring it inline with GitHub requirements. 

## Changes

This updates:

- `action.yml node` - Updates from `12` -> `16`
- `actions/checkout` - Update from `v1` & `v2` -> `v3`
- `types/node` - Updates from `12.19.9` -> `20.1.7n`
- `typescript` - Updates from `3.9.7` -> `4.8.2`
- `action/core` - Updates from `1.2.6` -> `1.10.0`

## Testing

- [x] Integration tests pass: [Build/Test#12](https://github.com/djthornton1212/ms-teams-notification/actions/runs/5004417166)

## Routine

- [x] Self-review: I have performed a review of my own code
- [ ] Documentation: I have updated javaDocs and `docs/` (if required) to reflect changes in this PR
- [x] Jira->PR Link: I have linked to this Pull Request from my Jira task through the GitHub <> Jira Integration
- [x] History: I have rebased to provide a clean and meaningful history

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -6,51 +6,59 @@ on: # rebuild any PRs and main branch changes
...
-      with: 
+      with:
         github-token: ${{ github.token }}
         ms-teams-webhook-uri: ${{ secrets.MS_TEAMS_WEBHOOK_URI }}
         notification-summary: "Emojify! &#x1F6A2​​ &#x2705;"
-        notification-color: 28a745 
-        timezone: America/Denver 
\ No newline at end of file
+        notification-color: 28a745
+        timezone: America/Denver
```
Comment (anuriyev) [inline .github/workflows/test.yml:64]: why Denver?

**Review by anuriyev — APPROVED** [review]
LGTM

File: .github/workflows/test.yml (hunk 1 above)
Comment (djthornton1212) [reply .github/workflows/test.yml:64]: It's from the originator's test. This is a fork of that repo. I haven't touched the tests. It's probably their timezone.

Comment (modunjo7) [reply .github/workflows/test.yml:64]: I guess it can be any time zone you set

**Review by modunjo7 — APPROVED** [review]

**Pull request merged**